### PR TITLE
feat/room-music-player

### PR DESCRIPTION
**Pull request by MAXOUXAX**

### CODE REVIEW

**Comment by MAXOUXAX** [comment]
@datsmysong/developpeurs hey, this is ready for review :)
would love it if you could do it before the end of the week

File: backend/src/musicplatform/remotes/SpotifyRemote.ts [hunk 1]
```diff
@@ -28,18 +29,22 @@ export default class SpotifyRemote extends Remote {
...
     const { access_token, expires_in, refresh_token } =
       data.user_profile.bound_services[0];
 
-    if (access_token === null || expires_in === null || refresh_token === null)
-      return null;
+    if (!access_token || !expires_in || !refresh_token) return null;
 
     const expiresIn = parseInt(expires_in);
 
+    // TODO: https://github.com/spotify/spotify-web-api-ts-sdk/issues/79
```
Comment (Flotss) [inline backend/src/musicplatform/remotes/SpotifyRemote.ts:47]: DO NOW this !?

File: backend/src/socketio/RoomIO.ts [hunk 2]
```diff
@@ -1,20 +1,12 @@
...
+import Room, { TypedSocket } from "./Room";
 
 const roomStorage = RoomStorage.getRoomStorage();
 
 export default function RoomIO(
-  socket: Socket<
-    ClientToServerEvents,
-    ServerToClientEvents
-  > /*, next: (err?: ExtendedError) => void*/
+  socket: TypedSocket /*, next: (err?: ExtendedError) => void*/
```
Comment (Flotss) [inline backend/src/socketio/RoomIO.ts:9]:
```
/*, next: (err?: ExtendedError) => void*/
```
unecessary comment no ?

what's the purpose of comment ?

File: expo/components/player/RoomPlayer.tsx [hunk 3]
```diff
@@ -116,7 +74,7 @@ const RoomPlayer: React.FC<RoomPlayerProps> = ({ room, socket }) => {
       </View>
 
       <Button type="outline" onPress={playCoolSong}>
-        play Duke & Jones - Call Me (Chill Mix)
+        play Martin Garrix & Lloyiso - Real Love
```
Comment (Flotss) [inline expo/components/player/RoomPlayer.tsx:77]: SO CRAZY

File: expo/components/player/SoundCloudPlayer.tsx [hunk 4]
```diff
@@ -128,25 +139,33 @@
...
         setResolveCurrentState(undefined);
       }
     }
   };
 
-  const getPlaybackState = (): Promise<PlayingJSONTrack> => {
-    return new Promise((resolve) => {
+  async function getPlaybackState(): Promise<
+    Response<PlayingJSONTrack | null>
+  > {
```
[the remark this answers is not in the document]
Comment (Flotss) [reply expo/components/player/SoundCloudPlayer.tsx:152]: Oups `Expected indentation of 4 spaces but found 2.`
Resolve this pls

File: expo/components/player/SoundCloudPlayer.tsx [hunk 5]
```diff
@@ -128,25 +139,33 @@ const SoundCloudPlayer = forwardRef<
...
       };
       sendMessage(command);
     });
-  };
-  const next = async () => {};
+  }
+
+  async function next(): Promise<Response<void>> {
+    return { error: null, data: undefined };
+  }
```
Comment (Flotss) [inline expo/components/player/SoundCloudPlayer.tsx:164]: we can't use it rn ?

**Comment by MAXOUXAX** [comment]
@ngomesodent @TheRealSharKzy @drmolixcool still awaiting your reviews guys 🤞

**Review by Flotss — APPROVED** [review]

**Review by TheRealSharKzy — APPROVED** [review]
LGTM, didn't see anything that shocked me...

**Review by drmolixcool — APPROVED** [review]

**Pull request merged**